### DISCUSSION TITLE
[Tempo] Handle Clockskew in Frontend

### DISCUSSION THREAD

**Discussion by joe-elliott** [grafana/grafana#39912] [Observability]
Sometimes a machine's clock can drift which can cause some spans to be misaligned with their parents/children. See example below:

![image](https://user-images.githubusercontent.com/2272392/135622850-07054698-f79e-46d8-a4eb-5771446b9939.png)

In Jaeger this can be [handled by the backend](https://github.com/jaegertracing/jaeger/blob/master/model/adjuster/clockskew.go), but Tempo is a natively multitenant system and whether or not clock skew should be fixed is [up for debate](https://github.com/jaegertracing/jaeger/issues/1459).

We should add the ability to either configure at a datasource level or have a toggle button per trace to remove the clockskew and realign spans with their parents. This way individual tenants or users can view both the raw trace as well as the adjusted trace. Additionally the adjusted span should indicate somewhere that it has been adjusted and by how much.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
